src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let resultObject = { ...state };
+
+  for (const obj of actions) {
+    switch (obj.type) {
+      case 'addProperties':
+        Object.assign(resultObject, obj.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const prop of obj.keysToRemove) {
+          delete resultObject[prop];
+        }
+        break;
+
+      default:
+        resultObject = {};
+        break;
+    }
+    result.push({ ...resultObject });
```

Comment:
```suggestion
    }
    
    result.push({ ...resultObject });
```
I prefer to use an empty line between different block codes.